zndraw/static/World/systems/select.js

```diff
@@ -273,14 +273,25 @@ class Selection {
                 .sub(transform_object.position)
                 .multiplyScalar(0.5)
                 .add(transform_object.position);
+
+              const point = this.line3D.addPoint(new_pos, index);
+              this.transform_controls.detach();
+              this.transform_controls.attach(point);
             } else {
-              // No previous point, add the point at the same position
-              new_pos = transform_object.position.clone();
+              if (this.line3D.anchorPoints.children.length > 1) {
```

Comment:
Can we add another condition, that if there is only one point, the duplicate is added at a slight offset to the current one? Right now if I press `d` when there is only one point nothing happens. Or do you think this is the expected behaviour?

Reply:
We would have to ensure, that the offset is outside of any neighboring particles. I would prefer leaving it as it is.